File: c/claude/p19_fc.c
#include <stdio.h>
#include <stdlib.h>
#include <errno.h>
#include <limits.h>

int main(int argc, char *argv[]) {
    if (argc < 2) {
        fprintf(stderr, "Usage: %s <integer>\n", argv[0]);
        return EXIT_FAILURE;
    }

    char *endptr;
    errno = 0;
    long value = strtol(argv[1], &endptr, 10);

    if (endptr == argv[1] || *endptr != '\0' || errno == ERANGE) {
        fprintf(stderr, "Error: \"%s\" is not a valid integer.\n", argv[1]);
        return EXIT_FAILURE;
    }

    /* Overflow check before adding 1000 */
    if (value > LONG_MAX - 1000) {
        fprintf(stderr, "Error: adding 1000 to %ld would overflow.\n", value);
        return EXIT_FAILURE;
    }

    long result = value + 1000;
    printf("%ld\n", result);
    return EXIT_SUCCESS;
}

File: javascript/claude/p19_fc.js
'use strict';

const args = process.argv.slice(2);

if (args.length < 1) {
    console.error(`Usage: node p19_fc.js <integer>`);
    process.exit(1);
}

const raw = args[0];
const parsed = Number(raw);

/* Reject non-integers, NaN, Infinity, and empty strings */
if (!Number.isInteger(parsed) || raw.trim() === '') {
    console.error(`Error: "${raw}" is not a valid integer.`);
    process.exit(1);
}

/* Overflow check: Number.MAX_SAFE_INTEGER is 2^53 - 1 */
if (parsed > Number.MAX_SAFE_INTEGER - 1000) {
    console.error(`Error: adding 1000 to ${parsed} exceeds safe integer range.`);
    process.exit(1);
}

const result = parsed + 1000;
console.log(result);
